assets/blocks/course-categories-block/hooks/use-colors/index.js

```diff
@@ -0,0 +1,69 @@
+/**
```

Comment:
I think that the `withColorSettings` hook may already handle this.

Reply:
Yes, it makes a very similar approach, but I found a bug that I was not able to reproduce outside of this component.
Using that hook, the editor was always saving one version before, so for example if I have:

Categories: Change to Grey > Change to Yellow > Change to Blue > Save
The editor was storing "Yellow" instead of Blue. 

I suspect it was happening only on this block, because it requires saving the colors to apply to the inner elements, instead of the root, breaking the hook.

After a lot of tries, I only have success using directly the Gutenberg hook.

Reply:
I was mostly interested in knowing what the issue was with `withColorSettings`. Since you've already determined that it doesn't work well in this block, I don't think it makes sense to revisit it again.

Ultimately, it would be nice to be able to use the same approach everywhere. Maybe when we're able to [refactor the existing functionality](https://github.com/Automattic/sensei/pull/5455#discussion_r947063486), it will help.

Reply:
Hi Gabriel, just out of curiosity I tried with `withColorSettings` and `compose` and also removed the `Settings` component from the `return` statement, I'm not sure but I think it worked for me. I've attached a video. Maybe I'm not doing everything right, but can you kindly check? Thank you :)

https://user-images.githubusercontent.com/6820724/185210937-ba8b5022-3d90-4843-b11c-a51dced01118.mov

Reply:
@donnapep  and @Imran92  I updated the PR with these two commits  [f03cf42](https://github.com/Automattic/sensei/pull/5455/commits/f03cf4236693ae4b9b39b3e3fa199aaa1162881c)  
and [a7ee53b] (https://github.com/Automattic/sensei/pull/5455/commits/a7ee53b0da7c809c32b94f3e4d11e84ec58789dd)

I tried to use both HOCs previously but I was having issues with the saving process, but now that the saving is handled by the hook, was possible to use the HOC without issues.

Sorry about the backing and forward, I was a bit confused with the block data flow, so I was not able to see it before 😔